x-pack/plugins/security_solution/public/common/hooks/flyout/use_sync_flyout_url_param.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { ALERTS_PATH } from '../../../../common/constants';
+import { useUpdateUrlParam } from '../../utils/global_query_string';
+import { TableId } from '../../../../common/types';
+import { useShallowEqualSelector } from '../use_selector';
+import { URL_PARAM_KEY } from '../use_url_state';
+import { dataTableActions, dataTableSelectors } from '../../store/data_table';
+import { tableDefaults } from '../../store/data_table/defaults';
+import type { FlyoutUrlState } from './types';
+
+export const useSyncFlyoutUrlParam = () => {
+  const updateUrlParam = useUpdateUrlParam<FlyoutUrlState>(URL_PARAM_KEY.flyout);
+  const { pathname } = useLocation();
+  const dispatch = useDispatch();
+  const getDataTable = dataTableSelectors.getTableByIdSelector();
+
+  // Only allow the alerts page for now to be saved in the url state
+  const { expandedDetail } = useShallowEqualSelector(
+    (state) => getDataTable(state, TableId.alertsOnAlertsPage) ?? tableDefaults
+  );
+
+  useEffect(() => {
+    const isOnAlertsPage = pathname === ALERTS_PATH;
```

Comment:
I wonder if it might make sense to use `isOnAlertsPage` that is returned from `useGetAlertDetailsFlyoutLink` here.